### PR TITLE
Added basic hapi server

### PR DESCRIPTION
**Pull request by nicholaschun**

### CODE REVIEW

File: src/backend/index.js [hunk 1]
```diff
@@ -1,6 +1,35 @@
...
+    if(shouldStart){
+        await server.start();
+        console.log('Server running on ', server.info.uri);
+    }
+};
+process.on('unhandledRejection', (err) => {
+    console.error(err);
+    process.exit(1);
+});
+export default init
```
Comment (aduwillie) [inline src/backend/index.js]: Missing semicolon

File: package.json [hunk 2]
```diff
@@ -4,8 +4,8 @@
   "description": "Social bookmarking service",
   "main": "src/backend/index.js",
   "scripts": {
-    "dev": "./node_modules/.bin/nodemon --exec 'babel-node ./src/backend/index.js'",
-    "start": "babel-node ./src/backend/index.js",
+    "dev": "./node_modules/.bin/nodemon --exec 'babel-node ./server.js'",
```
Comment (aduwillie) [inline package.json]: Perhaps I couldn't explain myself better. The `server.js` file should exist under the `backend` directory. The trick is that the `server.js` should define the server (I mean the init function) and the index.js file should pass true after calling the `init` function.

File: server.js [hunk 3]
```diff
@@ -0,0 +1,3 @@
+import init from './src/backend/index'
+const shouldStart = true
```
Comment (aduwillie) [inline server.js]: This file should be `index.js` instead. Also there is no need for this variable as the value `true` can be passed directly into the `init` function call.

File: src/backend/index.js [hunk 4]
```diff
@@ -1,6 +1,35 @@
...
-
-http.createServer((_, res) => {
-    res.write('Blinkas web server');
-    res.end();
-}).listen(3000);
+/* eslint-disable */
+import Hapi from '@hapi/hapi';
+import pug from 'pug';
+import Vision from '@hapi/vision';
+import constants from './../../constants'
```
Comment (aduwillie) [inline src/backend/index.js]: Could we separate the absolute imports from the relative imports using a newline? This makes code cleaner and readable.

File: package.json (hunk 2 above)
Comment (nicholaschun) [reply package.json]: so should i  keep both the `index.js` and `server.js` in the same directory that is the backend folder? or i should keep the `index.js` in the root directory

File: src/backend/server.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    });
+    if(shouldStart){
+        await server.start();
+        console.log('Server running on ', server.info.uri);
+    }
+};
+process.on('unhandledRejection', (err) => {
+    console.error(err);
+    process.exit(1);
+})
```
Comment (aduwillie) [inline src/backend/server.js]: Let's be consistent with the use of semi-colons. If needed, we can enforce it with `eslint`.

File: src/backend/server.js [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+process.on('uncaughtException', (err) => {
+    console.error(err);
+    process.exit(1);
+})
```
Comment (aduwillie) [inline src/backend/server.js]: Similar comment with sem-colons

**Pull request merged**